analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -330,6 +330,23 @@ def construct_config_handler(cls, args, context):
                 if m.group('analyzer') == cls.ANALYZER_NAME:
                     analyzer_config[m.group('key')] = m.group('value')
 
+        # If both --analyzer-config and -config (in --tidyargs) is given then
+        # these need to be merged. Since "HeaderFilterRegex" has a default
+        # value in --analyzer-config, we take --tidyargs stronger so user can
+        # overwrite its value.
+        for i, extra_arg in enumerate(handler.analyzer_extra_arguments):
+            if extra_arg.startswith('-config'):
+                if extra_arg == '-config':
+                    arg = handler.analyzer_extra_arguments[i + 1]
+                    arg_num = 2
+                else:
+                    arg = extra_arg[len('-config='):]
+                    arg_num = 1
+
+                analyzer_config.update(json.loads(arg))
+                del handler.analyzer_extra_arguments[i:i + arg_num]
+                break
```

Comment:
In python I always like to save indentation where it is possible. Here we can do the following thing:
```suggestion
            if not extra_arg.startswith('-config'):
                continue

            if extra_arg == '-config':
                arg = handler.analyzer_extra_arguments[i + 1]
                arg_num = 2
            else:
                arg = extra_arg[len('-config='):]
                arg_num = 1

            analyzer_config.update(json.loads(arg))
            del handler.analyzer_extra_arguments[i:i + arg_num]
            break
```